src/app/app.component.spec.ts

```diff
@@ -1,34 +1,65 @@
-import { TestBed, async } from '@angular/core/testing';
+import { TestBed, ComponentFixture, inject } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { Ng2PageScrollModule } from 'ng2-page-scroll';
 import { Http, HttpModule } from '@angular/http';
+import { Ng2PageScrollModule } from 'ng2-page-scroll';
+import { CookieService } from 'ngx-cookie-service';
 
 import { AppComponent } from './app.component';
-import { CookieService } from 'ngx-cookie-service';
-import { Api } from './services/api';
 
+import { Api } from './services/api';
 
 describe('AppComponent', () => {
-  beforeEach(async(() => {
+  let component: AppComponent;
+  let fixture: ComponentFixture<AppComponent>;
+  let cookieStatus: string;
+
+  const mockCookieService = {
+    get: (cookie) => cookieStatus,
+    set: (cookie, status) => null
+  };
+
+  beforeEach(() => {
     TestBed.configureTestingModule({
       providers: [
-        CookieService,
+        { provide: CookieService, useValue: mockCookieService },
         Api
       ],
       declarations: [
         AppComponent,
       ],
       imports: [
         RouterTestingModule,
-        Ng2PageScrollModule.forRoot(),
-        HttpModule
+        HttpModule,
+        Ng2PageScrollModule.forRoot()
       ]
     }).compileComponents();
-  }));
+  });
+
```

Comment:
Can we add some constructor tests for this class?
Also, looking at the real constructor, lines 26 and 27 set the same variable? Not necessarily related to this, but if you add a few tests, I would also fix that.

Reply:
Done